python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/abstract_interp_lib_gen.py

```diff
@@ -2710,6 +2713,17 @@ def aten〇leaky_relu〡dtype(self_rank_dtype: Tuple[int, int], negative_slope:
     dtypes = [self_dtype, negative_slope_dtype]
     return promote_dtypes(ranks, dtypes)
 
+@check_dtype_function(
+    _check_tensors_with_the_same_dtype(num_of_tensors=1, error_types={torch.bool}, alpha=1, scale=1, input_scale=2) +
+    _check_tensors_with_the_same_dtype(num_of_tensors=1, error_types={torch.bool, torch.int8, torch.uint8, torch.int16, torch.int32, torch.int64}, alpha=1.0, scale=1.0, input_scale=2.0))
+def aten〇elu〡dtype(self_rank_dtype: Tuple[int, int], alpha: Union[int, float, complex] = 1, scale: Union[int, float, complex] = 1, input_scale: Union[int, float, complex] = 1) -> int:
+    self_rank, self_dtype = self_rank_dtype
+    assert self_dtype != torch.bool
+    param_dtypes = [get_dtype_of_scalar(p) for p in [alpha, scale, input_scale]]
+    if any([is_float_dtype(d) for d in param_dtypes]):
+        assert not is_integer_dtype(self_dtype)
```

Comment:
I think this is not required.

Reply:
Without this logic we don't pass the `check_dtype` tests due to the fact that the op fails in certain cases where dtypes [can not be properly promoted](https://github.com/pytorch/pytorch/blob/256fed02e930210dbcd7e5e23fcf142362098c2a/torch/_refs/nn/functional/__init__.py#L231C34-L231C34) from a weaker type to a stronger type. Do you still want me to remove this logic? I also notice that similar logic is present in the `leaky_relu` dtype function, also mirroring the corresponding pytorch logic [here](https://github.com/pytorch/pytorch/blob/256fed02e930210dbcd7e5e23fcf142362098c2a/torch/_refs/nn/functional/__init__.py#L335).

Reply:
Thanks for the context!